app/src/main/java/com/github/dedis/student20_pop/ui/OrganizerFragment.java

```diff
@@ -0,0 +1,378 @@
+    private ExpandableListView expandableListView;
+    private Lao lao;  //should be given from intent or previous fragment
+    private Button propertiesButton;
+
+    /**
+     * Enum class for each event category
+     */
+    private enum EventCategory {
+        PAST, PRESENT, FUTURE
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+
+        //for testing purposes:
+        lao = new Lao("LAO I just joined", new Date(), new Keys().getPublicKey());
+
+        View rootView = inflater.inflate(R.layout.fragment_organizer, container, false);
+
+        //Display Events
+        expandableListView = rootView.findViewById(R.id.organizer_expandable_list_view);
+        listViewEventAdapter = new OrganizerExpandableListViewEventAdapter(this.getActivity(), getEvents(), true);
+        expandableListView.setAdapter(listViewEventAdapter);
+        expandableListView.expandGroup(0);
+        expandableListView.expandGroup(1);
+
+        //Display Properties
+        View properties = rootView.findViewById(R.id.properties_view);
+        ((TextView) properties.findViewById(R.id.organization_name)).setText(lao.getName());
+        ((TextView) properties.findViewById(R.id.witness_list)).setText("Witnesses: [id, id, id]");
+
+        propertiesButton = rootView.findViewById(R.id.tab_properties);
+        propertiesButton.setOnClickListener(
+                clicked -> {
+                    properties.setVisibility((properties.getVisibility() == View.GONE) ? View.VISIBLE : View.GONE);
+                });
+
+        return rootView;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    private List<Event> getEvents() {
+        /*
+        //Later:
+        List<String> eventsIds = lao.getEvents();
+        events = new ArrayList<>();
+        for (String id: eventsIds){
+            events.add(???.getEventFromId(id))
+        }
+         */
+
+        //Now (for testing) :
+        ArrayList<Event> events = new ArrayList<>();
+        events.add(new Event("Past Event 1", new Date(10 * 1000L), new Keys().getPublicKey(), "EPFL", "Poll"));
+        events.add(new Event("Past Event 2", new Date(20 * 1000L), new Keys().getPublicKey(), "CE-6", "Meeting"));
+        events.add(new Event("Present Event 1", new Date(500 * 1000L),
+                new Keys().getPublicKey(), "Geneva", "Roll-Call"));
+        events.add(new Event("Present Event 2", new Date(600 * 1000L),
+                new Keys().getPublicKey(), "Lausanne", "Discussion"));
+        events.add(new Event("Future Event 1", new Date(5000 * 1000L),
+                new Keys().getPublicKey(), "i don't know where yet", "Poll"));
+
+        return events;
+    }
+
+    public void launchEventCreationFragment(int eventType) {
+        switch (eventType) {
+            case 0:
```

Comment:
Is it possible to have an enum with the types? How do we know that 0 is for Meeting Event, 1 is for Roll-Call, and so on?

Reply:
I tried something, tell me I you prefer it that way

Reply:
It's better, thanks!